Code/src/main/resources/ui/FunctionBlock.fxml

```diff
@@ -1,28 +1,20 @@
-<?import nl.utwente.group10.ui.components.*?>
 <?import javafx.scene.control.*?>
 <?import javafx.scene.layout.*?>
-<?import java.net.*?>
 
-
-<FunctionBlock xmlns:fx="http://javafx.com/fxml/1" styleClass="block">
-	<BorderPane id="foreground" minHeight="90" minWidth="180">
+<fx:root type="nl.utwente.group10.ui.components.FunctionBlock" xmlns:fx="http://javafx.com/fxml/1" styleClass="function, block">
+	<BorderPane minHeight="90" minWidth="180">
 		<top>
 			<BorderPane>
 				<top>
-					<Label id="label_function_name" text="Function name"
-						styleClass="functionBlockTitle" />
+					<Label text="${controller.name}" styleClass="title" />
```

Comment:
Notice the `"${controller.name}"` binding expression. Also notice that the label does not have an id.